### PR TITLE
Convert INFO sub-fields (CSQ) into variant annotation

### PR DESCRIPTION
**Pull request by poggio84**
EVA-513

### CODE REVIEW

File: src/test/java/uk/ac/ebi/eva/pipeline/jobs/steps/processor/VariantToVariantAnnotationProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        sourceEntry.setAttributes(attributes);
+
+        Map<String, VariantSourceEntry> sourceEntries = new HashMap<>();
+        sourceEntries.put(FILE_ID, sourceEntry);
+        variant.setSourceEntries(sourceEntries);
+
+        VariantAnnotation variantAnnotation = processor.process(variant);
+        List<ConsequenceType> consequenceTypes = variantAnnotation.getConsequenceTypes();
+
+        assertTrue(variantAnnotation.getHgvs().isEmpty());
```
Comment (poggio84) [inline src/test/java/uk/ac/ebi/eva/pipeline/jobs/steps/processor/VariantToVariantAnnotationProcessorTest.java]: Should be null?

File: src/main/java/uk/ac/ebi/eva/pipeline/model/Csq.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
+package uk.ac.ebi.eva.pipeline.model;
+
+import org.beanio.annotation.Field;
+import org.beanio.annotation.Record;
+
+/**
+ *
+ */
+@Record
+public class Csq {
```
Comment (poggio84) [inline src/main/java/uk/ac/ebi/eva/pipeline/model/Csq.java]: Fields with '//' are mapped. The flag should be removed!

File: src/main/java/uk/ac/ebi/eva/pipeline/model/Csq.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+package uk.ac.ebi.eva.pipeline.model;
+
+import org.beanio.annotation.Field;
+import org.beanio.annotation.Record;
+
+/**
+ * Models the CSQ field generated by VEP. See 'VCF output' in http://www.ensembl.org/info/docs/tools/vep/vep_formats.html
+ */
+@Record
+public class Csq {
```
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/pipeline/model/Csq.java]: I understand the fields inside CSQ are configurable, would this implementation support that use case?

File: src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/processors/VariantToVariantAnnotationProcessor.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    private ConsequenceType csqToConsequenceTypeMapper(Csq csq) {
+        ConsequenceType consequenceType = new ConsequenceType();
+        consequenceType.setcDnaPosition(csq.getcDNAposition());
+        consequenceType.setCdsPosition(csq.getCdsPosition());
+        consequenceType.setBiotype(csq.getBiotype());
+        consequenceType.setEnsemblGeneId(csq.getGene());
+        consequenceType.setEnsemblTranscriptId(csq.getFeature());
+        consequenceType.setStrand(csq.getStrand());
+        consequenceType.setCodon(csq.getCodons());
+        consequenceType.setSoTermsFromSoNames(Arrays.asList(csq.getConsequence().split(",")));
```
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/pipeline/jobs/steps/processors/VariantToVariantAnnotationProcessor.java]: I feel we are creating many layers by forcing the dependency on Biodata. Given that we already have our own implementation of the VEP output reader and `VariantAnnotation` writer, I would suggest to expend some extra time on this and build a model that really suits us.

**Comment by cyenyxe** [comment]
This no longer applies.

**Pull request closed without merging**